plenario/api/point.py

```diff
@@ -1,120 +1,261 @@
-from plenario.database import session
-from plenario.models import MetaTable
+from plenario.api.validator import NoDefaultDatesValidator, validate, NoGeoJSONValidator, has_tree_filters, converters
+from plenario.api.jobs import make_job_response, submit_job, get_job, set_status, get_status, set_request, get_request, \
+    get_result, set_flag, get_flag
+from plenario.models import MetaTable, DataDump
+from plenario.database import fast_count, windowed_query
+
+# Use the standard pool if this is just the app,
+# but use the shared connection pool if this
+# is the worker. It's more efficient!
+if os.environ.get('WORKER'):
+    from worker import session
+else:
+    from plenario.database import session
 
 
 # ======
 # routes
 # ======
 
+# The get_job method in jobs.py does not have crossdomain, and is out of the
+# Flask context. So we define a wrapper here to access it.
+@crossdomain(origin="*")
+def get_job_view(ticket):
+    return get_job(ticket)
+
+
 @cache.cached(timeout=CACHE_TIMEOUT, key_prefix=make_cache_key)
 @crossdomain(origin="*")
 def timeseries():
     fields = ('location_geom__within', 'dataset_name', 'dataset_name__in',
-              'agg', 'obs_date__ge', 'obs_date__le', 'data_type')
+              'agg', 'obs_date__ge', 'obs_date__le', 'data_type', 'job')
     validator = NoGeoJSONValidator(only=fields)
-    validated_args = validate(validator, request.args.to_dict())
-    if validated_args.errors:
-        return bad_request(validated_args.errors)
+    validator_result = validate(validator, request.args.to_dict())
 
-    return _timeseries(validated_args)
+    if validator_result.errors:
+        return api_response.bad_request(validator_result.errors)
+
+    if validator_result.data.get('job'):
+        return make_job_response("timeseries", validator_result)
+    else:
+        panel = _timeseries(validator_result)
+        return api_response.timeseries_response(panel, validator_result)
 
 
 @cache.cached(timeout=CACHE_TIMEOUT, key_prefix=make_cache_key)
 @crossdomain(origin="*")
 def detail_aggregate():
     fields = ('location_geom__within', 'dataset_name', 'agg', 'obs_date__ge',
-              'obs_date__le', 'data_type')
+              'obs_date__le', 'data_type', 'job')
     validator = NoGeoJSONDatasetRequiredValidator(only=fields)
-    validated_args = validate(validator, request.args.to_dict())
-    if validated_args.errors:
-        return bad_request(validated_args.errors)
-    return _detail_aggregate(validated_args)
+    validator_result = validate(validator, request.args.to_dict())
+
+    if validator_result.errors:
+        return api_response.bad_request(validator_result.errors)
+
+    if validator_result.data.get('job'):
+        return make_job_response("detail-aggregate", validator_result)
+    else:
+        time_counts = _detail_aggregate(validator_result)
+        return api_response.detail_aggregate_response(time_counts, validator_result)
```

Comment:
If job is specified, make_job_response returns a lightweight ticket right away and submits a job corresponding to this endpoint.

In order to get at the raw information returned from completing an endpoint's work, we needed to separate out the step of converting a result into an HTTP response.

All of the endpoints follow this pattern.